anchors/dna/anchors/anchors.js

```diff
@@ -1,528 +1,99 @@
-    variables. But unlike global application variables in non-distributed environments, with anchors, 
-    the developer must consider and write appropriate validation rules for each anchor since they 
-    are actually stored as entries on the chain.
-
-    For details, see: https://github.com/Holochain/mixins/wiki/Anchors
-
-    Notes to add to docs:
-
-        - Can't use one call to removeFromList to remove multiple entries of same value but having different entry types
-*/
-
-/* Holocain API */ var _core_remove=remove;remove=function(a,b){return checkForError("remove",_core_remove(a,b))};var _core_makeHash=makeHash;makeHash=function(a,b){return checkForError("makeHash",_core_makeHash(a,b))};var _core_debug=debug;debug=function(a){return checkForError("debug",_core_debug(JSON.stringify(a)))};var _core_call=call;call=function(a,b,c){return __holochain_api_check_for_json(checkForError("call",_core_call(a,b,c)))};var _core_commit=commit;commit=function(a,b){return checkForError("commit",_core_commit(a,b))};var _core_get=get;get=function(a,b){return __holochain_api_check_for_json(checkForError("get",b===undefined?_core_get(a):_core_get(a,b)))};var _core_getLinks=getLinks;getLinks=function(a,b,c){return checkForError("getLinks",_core_getLinks(a,b,c))};var _core_send=send;send=function(a,b,c){return __holochain_api_check_for_json(checkForError("send",c===undefined?_core_send(a,b):_core_send(a,b,c)))};function __holochain_api_check_for_json(rtn){try{rtn=JSON.parse(rtn)}catch(err){}return rtn}function checkForError(func,rtn){if(typeof rtn==="object"&&rtn.name=="HolochainError"){var errsrc=new getErrorSource(4);var message='HOLOCHAIN ERROR! "'+rtn.message.toString()+'" on '+func+(errsrc.line===undefined?"":" in "+errsrc.functionName+" at line "+errsrc.line+", column "+errsrc.column);throw{name:"HolochainError",function:func,message:message,holochainMessage:rtn.message,source:errsrc,toString:function(){return this.message}}}return rtn}function getErrorSource(depth){try{throw new Error}catch(e){var line=e.stack.split("\n")[depth];var reg=/at (.*) \(.*:(.*):(.*)\)/g.exec(line);if(reg){this.functionName=reg[1];this.line=reg[2];this.column=reg[3]}}}
-
-// script globals
-var ANCHOR_GENERIC = "_anchor_generic_";
-var ENTRY_TYPE_PROP_NAME = "__ANCHORS_ENTRY_TYPE__";
-var ANCHOR_DELIMETER = "_-_anchor_delimeter_-_";
+/**
+ * @param anchor Object of the form {anchorType: <string>, anchorText: <string>}
+ * @return anchorHash
+ **/
+function anchor(anchor){
+  var anchorType = {anchorType: anchor.anchorType, anchorText: ''};
+  var rootAnchortype =  {anchorType: 'anchorTypes', anchorText: ''};
+  var anchorHash = makeHash('anchor', anchor);
+  var anchorGet = get(anchorHash);
+  if(anchorGet === null){
+    var anchorType = {anchorType: anchor.anchorType, anchorText: ''};
+    var rootAnchortype =  {anchorType: 'anchorTypes', anchorText: ''};
+    var anchorTypeGet = get(makeHash('anchor', anchorType));
```

Comment:
Why are these defined with 'var' again?

Reply:
Isn't that the standard way to do it in JS 5?

Reply:
looks like a bug.